cmd/fleetctl/generate_gitops.go

```diff
@@ -0,0 +1,1281 @@
+	return scriptSlice, nil
+}
+
+func (cmd *GenerateGitopsCommand) generatePolicies(teamId *uint, filePath string) ([]map[string]interface{}, error) {
+	policies, err := cmd.Client.GetPolicies(teamId)
+	if err != nil {
+		fmt.Fprintf(cmd.CLI.App.ErrWriter, "Error getting policies: %s\n", err)
+		return nil, err
+	}
+	if len(policies) == 0 {
+		return nil, nil
+	}
+	t := reflect.TypeOf(fleet.Policy{})
+	result := make([]map[string]interface{}, len(policies))
+	for i, policy := range policies {
+		policySpec := map[string]interface{}{
+			jsonFieldName(t, "Name"):                  policy.Name,
+			jsonFieldName(t, "Description"):           policy.Description,
+			jsonFieldName(t, "Resolution"):            policy.Resolution,
+			jsonFieldName(t, "Query"):                 policy.Query,
+			jsonFieldName(t, "Platform"):              policy.Platform,
+			jsonFieldName(t, "Critical"):              policy.Critical,
+			jsonFieldName(t, "CalendarEventsEnabled"): policy.CalendarEventsEnabled,
+		}
+		// Handle software automation.
+		if policy.InstallSoftware != nil {
+			if software, ok := cmd.SoftwareList[policy.InstallSoftware.SoftwareTitleID]; ok {
+				policySpec["install_software"] = map[string]interface{}{
+					"hash_sha256": software.Hash + " " + software.Comment,
+				}
+			} else {
+				policySpec["install_software"] = map[string]interface{}{
+					"hash_sha256": cmd.AddComment(filePath, "TODO: Add your hash_sha256 here"),
+				}
+				cmd.Messages.Notes = append(cmd.Messages.Notes, Note{
+					Filename: filePath,
+					Note:     fmt.Sprintf("Warning: policy %s software installer has no hash_sha256.  This is required for GitOps to work.  Please add it manually.", policy.Name),
+				})
+			}
+		}
+		// Handle script automation.
+		if policy.RunScript != nil {
+			if scriptPath, ok := cmd.ScriptList[policy.RunScript.ID]; ok {
+				policySpec["run_script"] = map[string]interface{}{
+					"path": scriptPath,
+				}
+			}
+		}
+		// Parse any labels.
+		if policy.LabelsIncludeAny != nil {
+			labels := make([]string, len(policy.LabelsIncludeAny))
+			for i, label := range policy.LabelsIncludeAny {
+				labels[i] = label.LabelName
+			}
+			policySpec["labels_include_any"] = labels
+		}
+		if policy.LabelsExcludeAny != nil {
+			labels := make([]string, len(policy.LabelsExcludeAny))
+			for i, label := range policy.LabelsExcludeAny {
+				labels[i] = label.LabelName
+			}
+			policySpec["labels_exclude_any"] = labels
+		}
+		result[i] = policySpec
+	}
+	return result, nil
+}
+
+func (cmd *GenerateGitopsCommand) generateQueries(teamId *uint) ([]map[string]interface{}, error) {
+	queries, err := cmd.Client.GetQueries(teamId, nil)
+	if err != nil {
+		fmt.Fprintf(cmd.CLI.App.ErrWriter, "Error getting queries: %s\n", err)
+		return nil, err
+	}
+	if len(queries) == 0 {
+		return nil, nil
+	}
+	t := reflect.TypeOf(fleet.Query{})
+	result := make([]map[string]interface{}, len(queries))
+	for i, query := range queries {
+		querySpec := map[string]interface{}{
```

Comment:
Missing `MinOsqueryVersion`?

Reply:
Hm.  These aren't documented in the YAML guide 😞 .  Will add.